### DISCUSSION TITLE
Using RYML as a library in a Visual Studio 2015 Solution

### DISCUSSION THREAD

**Discussion by jslezosky** [biojppm/rapidyaml#230] [Q&A] [answered]
I'm trying to use the ryml library in a VS 2015 x86 application.
I followed the vcpkg instructions from the "Adding ryml to Visual studio project. https://github.com/biojppm/rapidyaml/discussions/111" conversation.
The ryml package gets built, and is discoverable in Visual Studio, but when I include
<ryml/ryml_std.hpp>
<ryml/ryml.hpp>
<c4/ryml_std.hpp>
, the compiler does not recognize ryml::parse_in_arena().

What am I doing wrong?

**Comment by biojppm** [comment, marked as answer]
Looking at [the entry in vcpkg](https://vcpkg.info/port/ryml), the existing ryml version is old -- from october  2021 (or july, can't really tell), so it still does not have those functions. The vcpkg is [produced downstream](https://github.com/microsoft/vcpkg/tree/master/ports/ryml) and needs to be updated. I am not familiar with the details, as the package was put there by somebody else.

So you either need to push a package submission to the vcpkg, or compile ryml from source as the examples in the README show.
